src/Symfony/Component/FeatureToggle/Feature.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureToggle;
+
+use Symfony\Component\FeatureToggle\Strategy\StrategyInterface;
+
+final class Feature
+{
+    public function __construct(
+        private readonly string $name,
+        private readonly string $description,
+        private readonly bool $default,
+        private readonly StrategyInterface $strategy,
+    ) {
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getDescription(): string
+    {
+        return $this->description;
+    }
+
+    public function isEnabled(): bool
+    {
+        return $this->strategy->compute()->isEnabled($this->default);
+    }
```

Comment:
The Feature class looks more like a value object, so how about passing the strategy to the method as a collaborator and remove from constructor.

The `FeatureChecker` implementation can contain a mapping of feature name -> strategy service ID following [this comment](https://github.com/symfony/symfony/pull/51649/files#r1324812652) 

```suggestion
    public function isEnabled(StrategyInterface $strategy): bool
    {
        return $strategy->compute()->isEnabled($this->default);
    }
```

Reply:
If `FeatureChecker` contains the mapping, this code could be in 
```php
    public function isEnabled(string $featureName): bool
    {
        // Retrieve Feature and Strategy from mapping

        match ($strategy->compute()) {
            StrategyResult::Grant => true,
            StrategyResult::Deny => false,
            StrategyResult::Abstain => $feature->getDefault(),
        }
    }
```

The Feature object won't have any logic anymore.

Reply:
ATM the Feature object is meant to hold the strategy so you cannot instanciate it without it. If we move to some kind of mapping, then yes the logic would probably be in the FeatureChecker and would require to have some kind of "default strategy".